### PR TITLE
Add option to expand accordions to Manuals Publisher

### PR DESCRIPTION
**Pull request by callumknights**
Departmental request to add functionality to the Manuals format so that publishers
can choose to expand accordions, when expanded accordions are removed entirely.

The changes to the appearance of accordions is shown in this PR (https://github.com/alphagov/manuals-frontend/pull/1082)

image

Trello: https://trello.com/c/wgEKW9Ze/2398-5-allow-manuals-publisher-to-set-accordions-to-open

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: spec/controllers/sections_controller_spec.rb [hunk 1]
```diff
@@ -11,11 +11,20 @@
...
+        "title" => "title",
+        "summary" => "summary",
+        "body" => "body",
+        "change_note" => "change_note",
+        "minor_update" => true,
+        "visually_expanded" => false,
+      }
       expect(Section::CreateService).to receive(:new) { |args|
-        expect(args[:attributes].to_hash).to have_key(:title)
+        expect(args[:attributes].to_hash.keys).to eq(expected_keys)
```
Comment (brucebolt) [inline spec/controllers/sections_controller_spec.rb:24]: Nice 👍

File: app/views/sections/_form.html.erb [hunk 2]
```diff
@@ -11,6 +11,11 @@
       <div class="preview_button add-vertical-margins"></div>
       <div class="preview_container add-vertical-margins" style="display: none;"></div>
 
+      <div class="checkbox add-vertical-margins">
+        <%= f.check_box :visually_expanded, tag_type: :p, label: "Expand collapsible content(accordions)", checked_value: section.visually_expanded %>
```
Comment (brucebolt) [inline app/views/sections/_form.html.erb:15]:
```suggestion
        <%= f.check_box :visually_expanded, tag_type: :p, label: "Expand collapsible content (accordions)", checked_value: section.visually_expanded %>
```
Should there be a space here?

File: features/creating-and-editing-a-manual.feature [hunk 3]
```diff
@@ -64,6 +64,14 @@ Feature: Creating and editing a manual
     Given a draft manual exists without any sections
     When I create a section for the manual
     Then I see the manual has the new section
+    And I see the section is'nt visually expanded
```
Comment (brucebolt) [inline features/creating-and-editing-a-manual.feature:67]:
```suggestion
    And I see the section isn't visually expanded
```

**Review by brucebolt — COMMENTED** [review]
Looks good to me.  I've just left a couple of suggested changes for typos, which will also need updating in the corresponding tests.

File: spec/controllers/sections_controller_spec.rb (hunk 1 above)
Comment (callumknights) [reply spec/controllers/sections_controller_spec.rb:24]: 😄

File: features/creating-and-editing-a-manual.feature (hunk 3 above)
Comment (callumknights) [reply features/creating-and-editing-a-manual.feature]: 👍

File: app/views/sections/_form.html.erb (hunk 2 above)
Comment (callumknights) [reply app/views/sections/_form.html.erb]: 👍

**Review by brucebolt — APPROVED** [review]
Changes look good, but I see there's now end-to-end test failures related to publishing manuals.  I think that will need a fix, then you'll need to follow the [breaking change guide](https://github.com/alphagov/publishing-e2e-tests/blob/master/docs/breaking-app-change.md) to get the e2e tests to pass.

**Comment by callumknights** [comment]
> Changes look good, but I see there's now end-to-end test failures related to publishing manuals. I think that will need a fix, then you'll need to follow the [breaking change guide](https://github.com/alphagov/publishing-e2e-tests/blob/master/docs/breaking-app-change.md) to get the e2e tests to pass.

This is fixed by deploying this, content

> Changes look good, but I see there's now end-to-end test failures related to publishing manuals. I think that will need a fix, then you'll need to follow the [breaking change guide](https://github.com/alphagov/publishing-e2e-tests/blob/master/docs/breaking-app-change.md) to get the e2e tests to pass.

The e2e tests need to be ran with the manuals-publisher, content-schemas and manuals-frontend changes deployed in integration together. I've got them green on this PR before. I'll re-deploy everything when content have got back to me so I can merge.

**Review by brucebolt — APPROVED** [review]

**Pull request merged**